### PR TITLE
Gson interface

### PR DESCRIPTION
**Pull request by DeDiamondPro**
## Description

Add gson-interface, this allows you to register serializers and deserializers with ease
see: [https://github.com/mintern/gson-interface/blob/master/InterfaceExample.java](https://github.com/mintern/gson-interface/blob/master/InterfaceExample.java)

## Checklist
- [X] I made a clear description of what was changed
- [X] I stated why these changes were necessary
- [X] I updated documentation or said what needs to be updated
- [X] I made sure these changes are backwards compatible
- [X] This pull request is for one feature/bug fix

### CODE REVIEW

**Review by Wyvest — APPROVED** [review]

**Review by nxtdaydelivery — DISMISSED** [review]
lgtm yesyes

**Comment by nxtdaydelivery** [comment]
I won't merge because I assume you wanted moon to check licensing

File: src/main/java/cc/polyfrost/oneconfig/config/gson/gsoninterface/GsonContext.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
+package cc.polyfrost.oneconfig.config.gson.gsoninterface;
+
+/*
+ * This file is part of OneConfig.
+ * OneConfig - Next Generation Config Library for Minecraft: Java Edition
+ * Copyright (C) 2021, 2022 Polyfrost.
+ * Copyright (c) 2012, Brandon Mintern, EasyESI, Berkeley, CA
```
Comment (MoonTidez) [inline src/main/java/cc/polyfrost/oneconfig/config/gson/gsoninterface/GsonContext.java:7]: nice small (c)

Comment (MoonTidez) [reply src/main/java/cc/polyfrost/oneconfig/config/gson/gsoninterface/GsonContext.java:7]: also add the copyright for outside stuff (Brandon etc) to the `LICENSE NOTICE FOR ADAPTED CODE`

File: src/main/java/cc/polyfrost/oneconfig/config/gson/gsoninterface/JsonDeserialization.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
+package cc.polyfrost.oneconfig.config.gson.gsoninterface;
+
+/*
+ * This file is part of OneConfig.
+ * OneConfig - Next Generation Config Library for Minecraft: Java Edition
+ * Copyright (C) 2021, 2022 Polyfrost.
+ * Copyright (c) 2012, Brandon Mintern, EasyESI, Berkeley, CA
```
Comment (MoonTidez) [inline src/main/java/cc/polyfrost/oneconfig/config/gson/gsoninterface/JsonDeserialization.java:7]: see other comments

File: src/main/java/cc/polyfrost/oneconfig/config/gson/gsoninterface/InterfaceAdapterFactory.java [hunk 3]
```diff
@@ -0,0 +1,189 @@
+package cc.polyfrost.oneconfig.config.gson.gsoninterface;
+
+/*
+ * This file is part of OneConfig.
+ * OneConfig - Next Generation Config Library for Minecraft: Java Edition
+ * Copyright (C) 2021, 2022 Polyfrost.
+ * Copyright (c) 2012, Brandon Mintern, EasyESI, Berkeley, CA
```
Comment (MoonTidez) [inline src/main/java/cc/polyfrost/oneconfig/config/gson/gsoninterface/InterfaceAdapterFactory.java:7]: see other comments

**Review by MoonTidez — CHANGES_REQUESTED** [review]
see the comments
U put copyright stuff in the wrong place

**Review by Wyvest — APPROVED** [review]

**Review by nxtdaydelivery — APPROVED** [review]
lgtm

**Pull request merged**